File: src/prod/src/ServiceModel/reliability/failover/ReconfigurationType.cpp
// ------------------------------------------------------------
// Copyright (c) Microsoft Corporation.  All rights reserved.
// Licensed under the MIT License (MIT). See License.txt in the repo root for license information.
// ------------------------------------------------------------

#include "stdafx.h"

using namespace Common;

namespace Reliability
{
	namespace ReconfigurationType
	{
		void WriteToTextWriter(TextWriter & w, Enum const& value)
		{
			switch (value)
			{
				case Other:
					w << "Other";
					return;
				case SwapPrimary:
					w << "SwapPrimary";
					return;
				case Failover:
					w << L"Failover";
					return;
                case None:
                    w << L"None";
                    return;
				default:
					Assert::CodingError("unknown value for enum {0}", static_cast<int>(value));
					return;
			}
		}

		ENUM_STRUCTURED_TRACE(ReconfigurationType, Other, LastValidEnum);

		::FABRIC_RECONFIGURATION_TYPE Reliability::ReconfigurationType::ConvertToPublicReconfigurationType(ReconfigurationType::Enum toConvert)
        {
            switch (toConvert)
            {
            case Other:
                return ::FABRIC_RECONFIGURATION_TYPE_OTHER;
            case SwapPrimary:
                return ::FABRIC_RECONFIGURATION_TYPE_SWAPPRIMARY;
            case Failover:
                return ::FABRIC_RECONFIGURATION_TYPE_FAILOVER;
            case None:
                return ::FABRIC_RECONFIGURATION_TYPE_NONE;
            default:
                Common::Assert::CodingError("Unknown Reconfiguration Type");
            }
        }
	}
}

File: src/prod/src/managed/Api/src/System/Fabric/ReconfigurationType.cs
// ------------------------------------------------------------
// Copyright (c) Microsoft Corporation.  All rights reserved.
// Licensed under the MIT License (MIT). See License.txt in the repo root for license information.
// ------------------------------------------------------------

namespace System.Fabric
{
    using System.Fabric.Interop;

    /// <summary>
    /// Represents replica reconfiguration type.
    /// </summary>
    public enum ReconfigurationType
    {
        /// <summary>
        /// Invalid reconfiguration type.
        /// </summary>
        Unknown,

        /// <summary>
        /// Reconfiguration triggered to promote an active secondary replica to the primary role and demote existing primary replica to secondary role.
        /// </summary>
        SwapPrimary,

        /// <summary>
        /// Reconfiguration triggered in response to a primary going down. This could be due to many reasons such as primary replica crashing etc
        /// </summary>
        Failover,

        /// <summary>
        /// Reconfigurations where no change is happening to the primary.
        /// </summary>
        Other,

        /// <summary>
        /// No reconfiguration is taking place currently.
        /// </summary>
        None,
    }
}